plugin/helpers_bots.go

```diff
@@ -4,12 +4,81 @@
 package plugin
 
 import (
+	"io/ioutil"
+	"path/filepath"
+
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/pkg/errors"
 )
 
-func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error) {
+type EnsureBotOptions struct {
+	ProfileImagePath string
+	IconImagePath    string
+}
+
+type EnsureBotOption func(*EnsureBotOptions)
+
+func ProfileImagePath(path string) EnsureBotOption {
+	return func(args *EnsureBotOptions) {
+		args.ProfileImagePath = path
+	}
+}
+
+func IconImagePath(path string) EnsureBotOption {
+	return func(args *EnsureBotOptions) {
+		args.IconImagePath = path
+	}
+}
+
+func (p *HelpersImpl) readFile(path string) ([]byte, error) {
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get bundle path")
+	}
+
+	imageBytes, err := ioutil.ReadFile(filepath.Join(bundlePath, path))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read image")
+	}
+	return imageBytes, nil
+}
+
+func (p *HelpersImpl) setBotIconAndProfileImage(botId string, profileImagePath string, iconImagePath string) error {
```

Comment:
can we name this helper `setBotImages` instead? otherwise adding another image parameter would result in another update to the name of this method :P

Reply:
Also the convention we've been using to organize the code is to keep all the helpers at the bottom of the file, easier to read through the file quickly to see what the api is that way, as opposed to sift through and figure out the public apis when it's intermixed with helper functions.